tritium-metrics/src/main/java/com/palantir/tritium/event/metrics/TaggedMetricInvocationEventHandler.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.event.metrics;
+
+import static com.palantir.logsafe.Preconditions.checkNotNull;
+
+import com.codahale.metrics.Timer;
+import com.palantir.tritium.event.AbstractInvocationEventHandler;
+import com.palantir.tritium.event.DefaultInvocationContext;
+import com.palantir.tritium.event.InstrumentationProperties;
+import com.palantir.tritium.event.InvocationContext;
+import com.palantir.tritium.event.InvocationEventHandler;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.lang.reflect.Method;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * An implementation of {@link AbstractInvocationEventHandler} whose purpose is to provide tagged metrics as
+ * described in {@code invocation-metrics.yml}.
+ */
+public final class TaggedMetricInvocationEventHandler extends AbstractInvocationEventHandler<InvocationContext> {
```

Comment:
I'm assuming you'd wire this in downstream in WC?

Reply:
This is the piece I haven't ironed out yet, long term I'd prefer if `Tritium.instrument` used this instead of the existing tagged metrics. As an intermediate step we could report both, but I'm not sure how much impact that would have on dd.

Reply:
Updated to wire both for now.